### PR TITLE
Improvement xdebug varnish tunnel

### PR DESCRIPTION
**Pull request by pinkeen**

### CODE REVIEW

File: roles/cs.mageops-cli-user/templates/bashrc.d/debug-tunnel.bash [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  ---------------------------------------------------------
+
+  1. Set up a tunnel from your $(echo -en "\033[1mlocal computer\033[0m"):
+
+  $(echo -en "\033[0;35m")ssh -N \\
+    -o ExitOnForwardFailure=yes \\
+    -o GatewayPorts=yes \\
+    -R {{ php_fpm_debug_pool_xdebug_remote_port }}:localhost:{{ php_fpm_debug_pool_xdebug_remote_port }} \\
+      $(whoami)@$(curl -sL https://api.ipify.org)$(echo -en "\033[0m")
```
Comment (krisdante) [inline roles/cs.mageops-cli-user/templates/bashrc.d/debug-tunnel.bash:17]: Please extract this URL to a setting.
Please use http://169.254.169.254/latest/meta-data/public-ipv4 for AWS

File: group_vars/vagrant.yml [hunk 2]
```diff
@@ -3,12 +3,49 @@ mageops_app_host_group: vagrant
...
+  - 192.168.100.0/24    # Our VirtualBox / Vagrant Host-Only network
+  - 10.0.2.0/24         # Default VirtualBox / Vagrant NAT network
 
 mageops_packages_app_node_extra:
   - ghostscript
 
-firewall_enable: no
+unison_server_port: 5566
+
+firewall_enable: yes
```
[the remark this answers is not in the document]
Comment (pinkeen) [reply group_vars/vagrant.yml]: Removed from this MR.

File: roles/cs.mageops-cli-user/templates/bashrc.d/debug-tunnel.bash (hunk 1 above)
Comment (pinkeen) [reply roles/cs.mageops-cli-user/templates/bashrc.d/debug-tunnel.bash]: Fixed as agreed.

File: roles/cs.varnish/templates/vcl/subroutines/recv.vcl.j2 [hunk 3]
```diff
@@ -131,7 +131,7 @@ if (
 ) {
     set req.http.{{ varnish_debug_request_info_header_name }} = "Token-Verified; Cache-Bypassed; Xid=" + req.xid + "";
 
-    return (pipe);
+    return (pass);
```
Comment (pinkeen) [inline roles/cs.varnish/templates/vcl/subroutines/recv.vcl.j2:134]: Reversed a mistake. `(pipe)` caused all varnish logic to be bypassed and was not the best choice here.

File: group_vars/all.yml [hunk 4]
```diff
@@ -1629,7 +1636,23 @@ ntp_timezone: "{{ mageops_timezone }}"
...
-# Misc
-aws_autoscaling_triggers_list: []
-aws_autoscaling_lambda_functions_list: []
+# ------------------------
+# --------  SSHD  --------
+# ------------------------
+
+sshd_config_users: 
+  root: "{{ sshd_config_allow_all_forwards }}"
+  magento: "{{ sshd_config_allow_all_forwards }}"
```
Comment (pinkeen) [inline group_vars/all.yml:1645]: We need `magento` user to be able to set up reverse gateway port tunnel forward.

**Pull request merged**